### ISSUE TITLE
Agenda file x is not in `org-mode'

### ISSUE THREAD

**Issue by hello-code** [bastibe/org-journal#20]
When I type C-c a a, this message appear in minibuffer:
Agenda file /mnt/hgfs/iemacs/journal/20140707 is not in `org-mode'

There are some files like:20140707 20140708 20140709... In journal directory .
When I change file name to 20140707.org,the message show 20140708 is not in `org-mode'.When I change file name to 20140708.org,the message show 20140709 is not in`org-mode'...seem org-journal do not know file type.

**Comment by jetaggart** [comment]
If someone is seeing this using the spacemacs layer https://github.com/syl20bnr/spacemacs/tree/develop/layers/%2Bemacs/org#org-journal-support, you need to initialize everything after `org-journal` in the `dotspacemacs/user-config` block:

```
(defun dotspacemacs/user-config ()
  (with-eval-after-load 'org-journal
    (setq org-journal-dir "~/Dropbox/org/journal")
    (setq org-agenda-files (list "~/Dropbox/org/journal"))
    (setq org-agenda-file-regexp "\\`[^.].*\\.org\\'\\|\\`[0-9]+\\'")
  )
```

**Comment by kenogo** [comment]
@jetaggart Didn't work for me :-/ Did you set anything for `eval-after-load 'org` or when you included org-journal in the layers section?

**Comment by cslux** [comment]
Maybe related to #96.

**Comment by cslux** [comment]
@kenogo What's the status. Did you get it working?

**Comment by kenogo** [comment]
@cslux Sadly, no, I'm currently using a workaround by naming the files with a `.org` extension. This is explicitly advised against, though, as it breaks some functionality.

**Comment by cslux** [comment]
Could you please post your org-journal setup? Are you using spacemacs?

**Comment by kenogo** [comment]
@cslux Yes, I am. And I will tomorrow after getting some sleep. Thanks for your help!

**Comment by kenogo** [comment]
Sorry for the delay. This is my minimal config to reproduce:

```
;; -*- mode: emacs-lisp -*-
;; This file is loaded by Spacemacs at startup.
;; It must be stored in your home directory.

(defun dotspacemacs/layers ()
  (setq-default
   dotspacemacs-distribution 'spacemacs
   dotspacemacs-enable-lazy-installation 'unused
   dotspacemacs-ask-for-lazy-installation t
   dotspacemacs-configuration-layer-path '()
   dotspacemacs-configuration-layers
   '(
     (org :variables
          org-enable-org-journal-support t
          org-journal-dir "~/spacemacs-bug-test/bujo")
     )
   dotspacemacs-frozen-packages '()
   dotspacemacs-excluded-packages '()
   dotspacemacs-install-packages 'used-only))

(defun dotspacemacs/init ()
  (setq-default
   dotspacemacs-elpa-https t
   dotspacemacs-elpa-timeout 5
   dotspacemacs-check-for-update nil
   dotspacemacs-elpa-subdirectory nil
   dotspacemacs-editing-style 'vim
   dotspacemacs-mode-line-theme '(spacemacs :separator slant)
   dotspacemacs-verbose-loading nil
   dotspacemacs-startup-banner 2
   dotspacemacs-startup-lists '((agenda . 10)
                                (todos . 5))
   dotspacemacs-startup-buffer-responsive t
   dotspacemacs-scratch-mode 'org-mode
   dotspacemacs-themes '(spacemacs-dark
                         spacemacs-light)
   dotspacemacs-colorize-cursor-according-to-state t
   dotspacemacs-default-font '("Source Code Pro"
                               :size 13
                               :weight normal
                               :width normal
                               :powerline-scale 1.1)
   dotspacemacs-leader-key "SPC"
   dotspacemacs-emacs-command-key "SPC"
   dotspacemacs-ex-command-key ":"
   dotspacemacs-emacs-leader-key "M-m"
   dotspacemacs-major-mode-leader-key ","
   dotspacemacs-major-mode-emacs-leader-key "C-M-m"
   dotspacemacs-distinguish-gui-tab nil
   dotspacemacs-remap-Y-to-y$ nil
   dotspacemacs-retain-visual-state-on-shift t
   dotspacemacs-visual-line-move-text nil
   dotspacemacs-ex-substitute-global nil
   dotspacemacs-default-layout-name "Default"
   dotspacemacs-display-default-layout nil
   dotspacemacs-auto-resume-layouts nil
   dotspacemacs-large-file-size 1
   dotspacemacs-auto-save-file-location 'original
   dotspacemacs-max-rollback-slots 5
   dotspacemacs-helm-resize nil
   dotspacemacs-helm-no-header nil
   dotspacemacs-helm-position 'bottom
   dotspacemacs-helm-use-fuzzy 'always
   dotspacemacs-enable-paste-transient-state nil
   dotspacemacs-which-key-delay 0.4
   dotspacemacs-which-key-position 'bottom
   dotspacemacs-loading-progress-bar t
   dotspacemacs-fullscreen-at-startup nil
   dotspacemacs-fullscreen-use-non-native nil
   dotspacemacs-maximized-at-startup t
   dotspacemacs-active-transparency 90
   dotspacemacs-inactive-transparency 90
   dotspacemacs-show-transient-state-title t
   dotspacemacs-show-transient-state-color-guide t
   dotspacemacs-mode-line-unicode-symbols t
   dotspacemacs-smooth-scrolling t
   dotspacemacs-line-numbers '(:relative nil
                               :enabled-for-modes prog-mode
                               :size-limit-kb 1000)
   dotspacemacs-folding-method 'evil
   dotspacemacs-smartparens-strict-mode nil
   dotspacemacs-smart-closing-parenthesis nil
   dotspacemacs-highlight-delimiters 'all
   dotspacemacs-persistent-server nil
   dotspacemacs-search-tools '("ag" "pt" "ack" "grep")
   dotspacemacs-default-package-repository nil
   dotspacemacs-whitespace-cleanup nil
   ))

(defun dotspacemacs/user-init ()
  )

(defun dotspacemacs/user-config ()
  ;; ORG JOURNAL
  (with-eval-after-load 'org-journal
    (setq org-journal-dir "~/spacemacs-bug-test/bujo")
    (setq org-journal-file-format "%Y/%m/%d")
    (setq org-agenda-files (list "~/spacemacs-bug-test/bujo/2019/03"
                                 "~/spacemacs-bug-test/bujo/2019/04"
                                 "~/spacemacs-bug-test/bujo"))
    ;; This sadly doesn't work
    (setq org-agenda-file-regexp "\\`[^.].*\\.org\\'\\|\\`[0-9]+\\'")
    )
  )
```

I have my journal files in the directories `~/spacemacs-bug-test/bujo/2019/03` and `~/spacemacs-bug-test/bujo/2019/04`

**Comment by kenogo** [comment]
The entry added to the list `auto-mode-alist` seems to be missing the forward slashes between the dates, if I uderstood it properly...
```
("/home/keno/spacemacs-bug-test/bujo/\\(?1:[0-9]\\{4\\}\\)\\(?2:[0-9][0-9]\\)\\(?3:[0-9][0-9]\\)\\(\\.gpg\\)?\\'" . org-journal-mode)

**Comment by cslux** [comment]
@kenogo Exactly!

**Comment by cslux** [comment]
I'm trying it my self with spacemacs.

**Comment by cslux** [comment]
You must set some of the variables before loading org-journal, or use `custom-set-variables` or the customize interface for that. 

- org-journal-dir
- org-journal-file-format

**Comment by cslux** [comment]
You could use `org-journal-enable-agenda-integration` to let org-journal handle adding the journals to `org-agenda-files`.

**Comment by cslux** [comment]
Continuing tomorrow... :-)

**Comment by kenogo** [comment]
@cslux Not required to continue tomorrow, I got it! Thank you so much. All I needed to do was move the `org-journal-dir` and `org-journal-file-format` declarations to `dotspacemacs/layers`, and also set `org-journal-enable-agenda-integration` to `t` as you said:
```
(defun dotspacemacs/layers ()
  (setq-default
   dotspacemacs-configuration-layers
   '(
     (org :variables
          org-enable-org-journal-support t
          org-journal-dir "~/spacemacs-bug-test/bujo"
          org-journal-file-format "%Y/%m/%d"
          org-journal-enable-agenda-integration t
          )
     ))
```
Everything seems to work flawlessly with this setup. Finally I have calendar integration :D Thanks again, I think this can be closed

**Comment by cslux** [comment]
Perfect! Glad it's working now. Closing!

**Comment by doyougnu** [comment]
FYI for anyone coming here years later, I was getting this error (on spacemacs) and nothing in the thread helped including using `eval-with-after-load 'org-journal` and setting `:variables`. I ended up fixing it by just creating all journal entries with `.org` like this:

```
     (org :variables
           ...
          org-journal-file-format "%Y/%m/%d.org"   ;; notice the .org
          org-journal-skip-carryover-drawers (list "LOGBOOK")  ;; rest of this stuff didn't matter including the order of org-journal-enable-agenda-integration
          org-journal-enable-agenda-integration t
          )
```

This also broke `evil-escape` in spacemacs so I couldn't use `fd` to escape from modes. This is a downstream effect caused by an error thrown from `'(org-agenda-files ....` in `(custom-set-variables ...` in the `.spacemacs` file. So if you see this error in `*messages*`: `file x not in org-mode` then you can find the file in custom-set-variables and remove it and get your escape back.

**Comment by frederic-santos** [comment]
Hi! I was getting this error since a few days, and I'm *not* using Spacemacs 😄 I regularly got messages like this in the minibuffer, especially when calling my agenda:
```
error in process sentinel: async-handle-result: Agenda file xxxxx/20220901 is not in Org mode
error in process sentinel: Agenda file xxxxx/20220901 is not in Org mode
```

This was fixed simply by doing this, as in the previous post:

>  I ended up fixing it by just creating all journal entries with `.org`.

org-agenda does seem to expect the extension `.org` for all agenda files. Indeed, the current default value of `org-agenda-file-regexp` in Org 9.5.5 is `"\\[^.].*\\.org\\'"`. Thus, if `org-journal-enable-agenda-integration` is `t`, this should be taken into account (at least by org-journals users) in one way or another.

@bastibe and other maintainers: if I understood well, it might be better to give a new default value for `org-journal-file-format`, or at least to propose in the README to add
```
(setq org-journal-file-format "%Y%m%d.org")
```
whenever we set `org-journal-enable-agenda-integration` to `t`.

(I can propose a PR if you want.)
